### ISSUE TITLE
Zero-loss issue when training on KITTI pedestrians

### ISSUE THREAD

**Issue by joonjeon** [xiaohaoChen/rrc_detection#17]
Hi, there!

Aside from training and testing RRC for cars, I'm trying to train and test the model for pedestrians, too.
I see there are several relevant scripts for KITTI cars to do so, so I have just changed the `car` string to `pedestrian` string.

As far as I know, this should work well without errors,
but the training log is as follows:
```Shell
I0619 18:36:08.662021 21580 solver.cpp:231] Iteration 0, loss = 0
I0619 18:36:08.662055 21580 solver.cpp:247]     Train net output #0: mbox_loss = 0 (* 1 = 0 loss)
I0619 18:36:08.662075 21580 solver.cpp:247]     Train net output #1: mbox_loss2 = 0 (* 1 = 0 loss)
I0619 18:36:08.662080 21580 solver.cpp:247]     Train net output #2: mbox_loss3 = 0 (* 1 = 0 loss)
I0619 18:36:08.662083 21580 solver.cpp:247]     Train net output #3: mbox_loss4 = 0 (* 1 = 0 loss)
I0619 18:36:08.662087 21580 solver.cpp:247]     Train net output #4: mbox_loss5 = 0 (* 1 = 0 loss)
I0619 18:36:08.662096 21580 sgd_solver.cpp:106] Iteration 0, lr = 0.0005
I0619 18:38:15.238804 21580 solver.cpp:231] Iteration 10, loss = 0
I0619 18:38:15.238950 21580 solver.cpp:247]     Train net output #0: mbox_loss = 0 (* 1 = 0 loss)
I0619 18:38:15.238955 21580 solver.cpp:247]     Train net output #1: mbox_loss2 = 0 (* 1 = 0 loss)
I0619 18:38:15.238962 21580 solver.cpp:247]     Train net output #2: mbox_loss3 = 0 (* 1 = 0 loss)
I0619 18:38:15.238970 21580 solver.cpp:247]     Train net output #3: mbox_loss4 = 0 (* 1 = 0 loss)
I0619 18:38:15.238973 21580 solver.cpp:247]     Train net output #4: mbox_loss5 = 0 (* 1 = 0 loss)
I0619 18:38:15.238977 21580 sgd_solver.cpp:106] Iteration 10, lr = 0.0005
I0619 18:40:21.685006 21580 solver.cpp:231] Iteration 20, loss = 0
I0619 18:40:21.685132 21580 solver.cpp:247]     Train net output #0: mbox_loss = 0 (* 1 = 0 loss)
I0619 18:40:21.685137 21580 solver.cpp:247]     Train net output #1: mbox_loss2 = 0 (* 1 = 0 loss)
I0619 18:40:21.685142 21580 solver.cpp:247]     Train net output #2: mbox_loss3 = 0 (* 1 = 0 loss)
I0619 18:40:21.685145 21580 solver.cpp:247]     Train net output #3: mbox_loss4 = 0 (* 1 = 0 loss)
I0619 18:40:21.685149 21580 solver.cpp:247]     Train net output #4: mbox_loss5 = 0 (* 1 = 0 loss)
I0619 18:40:21.685153 21580 sgd_solver.cpp:106] Iteration 20, lr = 0.0005
I0619 18:42:29.448524 21580 solver.cpp:231] Iteration 30, loss = 0
I0619 18:42:29.448693 21580 solver.cpp:247]     Train net output #0: mbox_loss = 0 (* 1 = 0 loss)
I0619 18:42:29.448700 21580 solver.cpp:247]     Train net output #1: mbox_loss2 = 0 (* 1 = 0 loss)
I0619 18:42:29.448704 21580 solver.cpp:247]     Train net output #2: mbox_loss3 = 0 (* 1 = 0 loss)
I0619 18:42:29.448707 21580 solver.cpp:247]     Train net output #3: mbox_loss4 = 0 (* 1 = 0 loss)
I0619 18:42:29.448710 21580 solver.cpp:247]     Train net output #4: mbox_loss5 = 0 (* 1 = 0 loss)
I0619 18:42:29.448715 21580 sgd_solver.cpp:106] Iteration 30, lr = 0.0005

```
Zero-loss. That is the question.

I have a presentation related to this on Thursday this week.
So I have to get this working as soon as possible :(
Any ideas what's going wrong?

**Comment by joonjeon** [comment]
Oops!
For `Car`(i.e. the string with capital c), I mistakenly changed it to `pedestrian`(i.e. the string with small p), while it was supposed to be `Pedestrian`(i.e. the string with capital p). Haha.

I'll check and keep the training phase running to see if it works.
If no problem, I'll close this issue.

**Comment by joonjeon** [comment]
Seems like there is a problem...

with learning rate of 0.0005, the loss value diverges and `NaN` is printed out.
So I have changed the learning rate to 0.00004. But I am not sure this will help.

I'll keep track of the training, and I'd appreciate if you can give me the details on how the RRC have actually been trained for submission on KITTI.

Edit 1: still no help when learning rate is 0.00004.
```Shell
I0620 18:27:46.953780 30500 solver.cpp:231] Iteration 520, loss = 592.377
I0620 18:27:46.953908 30500 solver.cpp:247]     Train net output #0: mbox_loss = 5.46014 (* 1 = 5.46014 loss)
I0620 18:27:46.953914 30500 solver.cpp:247]     Train net output #1: mbox_loss2 = 5.06201 (* 1 = 5.06201 loss)
I0620 18:27:46.953919 30500 solver.cpp:247]     Train net output #2: mbox_loss3 = 5.26767 (* 1 = 5.26767 loss)
I0620 18:27:46.953923 30500 solver.cpp:247]     Train net output #3: mbox_loss4 = 5.38162 (* 1 = 5.38162 loss)
I0620 18:27:46.953927 30500 solver.cpp:247]     Train net output #4: mbox_loss5 = 5.34317 (* 1 = 5.34317 loss)
I0620 18:27:46.953933 30500 sgd_solver.cpp:106] Iteration 520, lr = 4e-05
I0620 18:30:39.289325 30500 solver.cpp:231] Iteration 530, loss = 889.704
I0620 18:30:39.289417 30500 solver.cpp:247]     Train net output #0: mbox_loss = 5.5147 (* 1 = 5.5147 loss)
I0620 18:30:39.289423 30500 solver.cpp:247]     Train net output #1: mbox_loss2 = 5.2226 (* 1 = 5.2226 loss)
I0620 18:30:39.289428 30500 solver.cpp:247]     Train net output #2: mbox_loss3 = 5.46305 (* 1 = 5.46305 loss)
I0620 18:30:39.289432 30500 solver.cpp:247]     Train net output #3: mbox_loss4 = 5.4184 (* 1 = 5.4184 loss)
I0620 18:30:39.289435 30500 solver.cpp:247]     Train net output #4: mbox_loss5 = 5.6185 (* 1 = 5.6185 loss)
I0620 18:30:39.289440 30500 sgd_solver.cpp:106] Iteration 530, lr = 4e-05
I0620 18:33:22.349851 30500 solver.cpp:231] Iteration 540, loss = nan
I0620 18:33:22.349990 30500 solver.cpp:247]     Train net output #0: mbox_loss = nan (* 1 = nan loss)
I0620 18:33:22.349997 30500 solver.cpp:247]     Train net output #1: mbox_loss2 = nan (* 1 = nan loss)
I0620 18:33:22.350014 30500 solver.cpp:247]     Train net output #2: mbox_loss3 = nan (* 1 = nan loss)
I0620 18:33:22.350020 30500 solver.cpp:247]     Train net output #3: mbox_loss4 = nan (* 1 = nan loss)
I0620 18:33:22.350024 30500 solver.cpp:247]     Train net output #4: mbox_loss5 = nan (* 1 = nan loss)
I0620 18:33:22.350028 30500 sgd_solver.cpp:106] Iteration 540, lr = 4e-05
I0620 18:36:01.397670 30500 solver.cpp:231] Iteration 550, loss = nan
I0620 18:36:01.397837 30500 solver.cpp:247]     Train net output #0: mbox_loss = nan (* 1 = nan loss)
I0620 18:36:01.397843 30500 solver.cpp:247]     Train net output #1: mbox_loss2 = nan (* 1 = nan loss)
I0620 18:36:01.397862 30500 solver.cpp:247]     Train net output #2: mbox_loss3 = nan (* 1 = nan loss)
I0620 18:36:01.397867 30500 solver.cpp:247]     Train net output #3: mbox_loss4 = nan (* 1 = nan loss)
I0620 18:36:01.397871 30500 solver.cpp:247]     Train net output #4: mbox_loss5 = nan (* 1 = nan loss)
I0620 18:36:01.397876 30500 sgd_solver.cpp:106] Iteration 550, lr = 4e-05
I0620 18:38:41.260489 30500 solver.cpp:231] Iteration 560, loss = nan
I0620 18:38:41.260613 30500 solver.cpp:247]     Train net output #0: mbox_loss = nan (* 1 = nan loss)
I0620 18:38:41.260619 30500 solver.cpp:247]     Train net output #1: mbox_loss2 = nan (* 1 = nan loss)
I0620 18:38:41.260624 30500 solver.cpp:247]     Train net output #2: mbox_loss3 = nan (* 1 = nan loss)
I0620 18:38:41.260628 30500 solver.cpp:247]     Train net output #3: mbox_loss4 = nan (* 1 = nan loss)
I0620 18:38:41.260632 30500 solver.cpp:247]     Train net output #4: mbox_loss5 = nan (* 1 = nan loss)
I0620 18:38:41.260637 30500 sgd_solver.cpp:106] Iteration 560, lr = 4e-05
I0620 18:41:19.845993 30500 solver.cpp:231] Iteration 570, loss = nan
I0620 18:41:19.846125 30500 solver.cpp:247]     Train net output #0: mbox_loss = nan (* 1 = nan loss)
I0620 18:41:19.846130 30500 solver.cpp:247]     Train net output #1: mbox_loss2 = nan (* 1 = nan loss)
I0620 18:41:19.846135 30500 solver.cpp:247]     Train net output #2: mbox_loss3 = nan (* 1 = nan loss)
I0620 18:41:19.846139 30500 solver.cpp:247]     Train net output #3: mbox_loss4 = nan (* 1 = nan loss)
I0620 18:41:19.846143 30500 solver.cpp:247]     Train net output #4: mbox_loss5 = nan (* 1 = nan loss)
I0620 18:41:19.846149 30500 sgd_solver.cpp:106] Iteration 570, lr = 4e-05
I0620 18:44:00.243125 30500 solver.cpp:231] Iteration 580, loss = nan
I0620 18:44:00.243229 30500 solver.cpp:247]     Train net output #0: mbox_loss = nan (* 1 = nan loss)
I0620 18:44:00.243235 30500 solver.cpp:247]     Train net output #1: mbox_loss2 = nan (* 1 = nan loss)
I0620 18:44:00.243239 30500 solver.cpp:247]     Train net output #2: mbox_loss3 = nan (* 1 = nan loss)
I0620 18:44:00.243243 30500 solver.cpp:247]     Train net output #3: mbox_loss4 = nan (* 1 = nan loss)
I0620 18:44:00.243248 30500 solver.cpp:247]     Train net output #4: mbox_loss5 = nan (* 1 = nan loss)
I0620 18:44:00.243252 30500 sgd_solver.cpp:106] Iteration 580, lr = 4e-05
I0620 18:46:49.078773 30500 solver.cpp:231] Iteration 590, loss = nan
I0620 18:46:49.078915 30500 solver.cpp:247]     Train net output #0: mbox_loss = nan (* 1 = nan loss)
I0620 18:46:49.078922 30500 solver.cpp:247]     Train net output #1: mbox_loss2 = nan (* 1 = nan loss)
I0620 18:46:49.078940 30500 solver.cpp:247]     Train net output #2: mbox_loss3 = nan (* 1 = nan loss)
I0620 18:46:49.078944 30500 solver.cpp:247]     Train net output #3: mbox_loss4 = nan (* 1 = nan loss)
I0620 18:46:49.078948 30500 solver.cpp:247]     Train net output #4: mbox_loss5 = nan (* 1 = nan loss)
I0620 18:46:49.078953 30500 sgd_solver.cpp:106] Iteration 590, lr = 4e-05
```

NaN issue occurs :(
Any ideas on why this is happening?

**Comment by joonjeon** [comment]
Well, I found out that `num_classes` value in python training script should be changed from `2` to `3`, since pedestrian label is enumerated with number `2`.

So, I'll close this self-solved issue. Hope this helps others who are struggling to test the RRC on pedestrians.
